### PR TITLE
[misc] Allow partial prefix benchmarking & random input generation for prefix benchmarking

### PR DESCRIPTION
**Pull request by rickyyx**
This PR
1. Allows a random generation for prompts when testing prefix caching
2. Allows controlling exact ratio of prefix sharing among requests (rather than fully cached requests like before)

### CODE REVIEW

**Comment by rickyyx** [comment]
Will run some more tests.

File: benchmarks/benchmark_prefix_caching.py [hunk 1]
```diff
@@ -77,68 +95,102 @@ def sample_requests(
...
-        if len(filtered_dataset) == num_requests:
+        if len(filtered_requests) == num_requests:
             break
 
         # Tokenize the prompts and completions.
-        prompt = dataset[i][0]
-        prompt_token_ids = tokenizer(prompt).input_ids
+        prompt_token_ids = prefix_token_ids + tokenizer(
+            dataset[i][0]).input_ids
+        prompt = tokenizer.decode(prompt_token_ids)
```
Comment (comaniac) [inline benchmarks/benchmark_prefix_caching.py:111]: It looks a bit weird to prepend a random prefix to a dataset. Should we make `prefix_len` only effective with random dataset?

**Review by comaniac — APPROVED** [review]
Otherwise LGTM

File: benchmarks/benchmark_prefix_caching.py (hunk 1 above)
Comment (rickyyx) [reply benchmarks/benchmark_prefix_caching.py:111]: yeah - makes sense to me too.

**Pull request merged**